src/jit/compiler.cpp

```diff
@@ -4442,13 +4442,6 @@ void Compiler::compCompile(void** methodCodePtr, ULONG* methodCodeSize, CORJIT_F
 #endif
 
 #ifndef LEGACY_BACKEND
-    // TODO-CQ: Remove "if-block" when lowering doesn't rely on front end liveness.
-    // Remove "if-block" to repro assert('!"We should never hit any assignment operator in lowering"')
-    // using self_host_tests_amd64\JIT\Methodical\eh\finallyexec\tryCatchFinallyThrow_nonlocalexit1_d.exe
-    if (!fgLocalVarLivenessDone)
-    {
-        fgLocalVarLiveness();
-    }
```

Comment:
There are no diffs in the executable code in, but there are diffs in the GC info for some methods when generating debuggable code. In particular, any variables that were zero-ref before `fgExtendDbgLifetimes` (and therefore zero-ref at the beginning of liveness) will go from being tracked to being untracked. This is because without these changes the local variable table gets resorted after the lifetimes of such variables have been extended. Because of the lifetime extension, these variables are no longer zero-ref at the start of liveness and the `lvaSortByRefCount` decides to make these variables tracked. With these changes, the local variable table does not get resorted after the lifetimes of the variables in question are extended, so `lvaSortByRefCount` only sees them as zero-ref and decides not to mark them as tracked.